src/stream.iced

```diff
@@ -0,0 +1,26 @@
+stream = require('stream')
+enc = require('./encoding')
+
+desired_block_size = 4096
+calculate_block_size = (input_length) ->
+  input_length*Math.floor(desired_block_size/input_length)
+
+exports.StreamEncoder = class StreamEncoder extends stream.Transform
+
+  constructor : (@encoder) ->
+    super({highWaterMark : calculate_block_size(encoder.in_block_len)})
+
+  _transform : (chunk, encoding, cb) ->
+    @push(@encoder.encode(chunk))
+    @read(0)
```

Comment:
Yes, but they don't consider it bad practice - it didn't hurt performance either. They key is this:
"If the internal read buffer is below the highWaterMark, and the stream is not currently reading, then calling stream.read(0) will trigger a low-level stream._read() call."
I think the reason for their "never need to do this" remark is that in most cases, it would be simpler to either wait for more data to come onto the stream, change the highWaterMark parameter, or write zeros until the highWaterMark is reached. We can't do any of those. I also think they were wrong: more than a few implementations of TransformStreams called `read(0)` in their `_transform` functions